### PR TITLE
Changed the ClusteringFactory to only do something, if there are any …

### PR DESCRIPTION
**Pull request by TwoOfTwelve**
There was an error, if all comparisons were beneath the threshold (-m in cli). Fixed by not executing clustering, if there are no comparisons left.

### CODE REVIEW

File: core/src/main/java/de/jplag/clustering/ClusteringFactory.java [hunk 1]
```diff
@@ -31,35 +31,39 @@ public class ClusteringFactory {
     private static final Logger logger = LoggerFactory.getLogger(ClusteringFactory.class);
 
     public static List<ClusteringResult<Submission>> getClusterings(Collection<JPlagComparison> comparisons, ClusteringOptions options) {
-        if (!options.enabled()) {
-            logger.warn(CLUSTERING_DISABLED);
-            return Collections.emptyList();
-        } else {
-            logger.info(CLUSTERING_PARAMETERS, options.algorithm(), options.preprocessor());
-        }
+        if (comparisons.isEmpty()) {
```
Comment (tsaglam) [inline core/src/main/java/de/jplag/clustering/ClusteringFactory.java:34]: Is that correct? Shouldn't that be `!empty`?

**Review by tsaglam — COMMENTED** [review]
Just one issue. Also, we could add a small unit test for the empty case.

**Review by tsaglam — APPROVED** [review]
